### ISSUE TITLE
[Feature Request] Allow option to hide RSS articles / Wallabag / wikipedia articles from history view

### ISSUE THREAD

**Issue by arooni** [koreader/koreader#6140]
* KOReader version: 2020.04.1
* Device: Paperwhite 4

I normally read books on my kindle, but when I feel like changing it up I use the very helpful Wallabag/RSS/Wikipedia features in KoReader.

One thing I notice after reading a few RSS articles or wallabag articles is that:

1.  The books that I normally want to read get buried in pages 2-3-4.  I'd suggest a way to apply a filter to the history view such that I can filter out articles, or perhaps choose what type of entries I want to see.

**Comment by mwoz123** [comment]
Not long time back I added this option to wallabag settings (see screenshot below) . It's for 100% read, marked as completed or abadoned.
For RSS you mean News Downloader or other source? 
For N.D. dir by default should be excluded from history. There's a FR to make it appear;)

![Zrzut ekranu z 2020-05-21 20-00-29](https://user-images.githubusercontent.com/4487025/82590545-2c625000-9b9e-11ea-93e8-3efe1bada144.png)

**Comment by arooni** [comment]
Ok that's super useful for wallabag.   My issue is I'll be lazy sometimes and read 10/11 pages because with wallabag sometimes it imports a bunch of html content not relevant to the actual article.  Maybe this is laziness since there's a way to mark the item as read or go to the end of the article.  

I do think being able to filter out news articles / wallabag articles would be useful to me (let's say i'm in progress with 10 walllabag / rss articles but want to get back to my book).  

Though I suppose there are various approaches to this including adding the book to my library.  Marking each item as finished manually to get them cleared out etc.  

Maybe if there was a way of marking the item as read via a multi gesture that would be helpful too.   

who knows, maybe i'm just being overly needy? :P

**Comment by Frenzie** [comment]
There's always book info → mark as read.

**Comment by arooni** [comment]
Perhaps letting the book be marked as read via a multi gesture ?

**Comment by ryanwwest** [comment]
I know this is old but it's a good idea - perhaps more generally than from a Wallabag/RSS-specific change, there could be a way to mark a specific folder as ignored from history (and maybe a similar option to mark a folder like `~/Downloads` as 'don't save metadata' as for random documents that I read once for a few minutes then delete, I don't need to create any metadata).

**Comment by Frenzie** [comment]
Marking a folder as ignored is an interesting idea.

**Comment by mergen3107** [comment]
Maybe using Tweak document settings?
Like I did with manga: if book is in a specific folder, then change direction to RTL. 

Here we can make a settings like "ignore_history" and make this check in History routine. 

Don't know how complicated this might be though :D

**Comment by ryanwwest** [comment]
@mergen3107 That sounds interesting. Is there anywhere I can read (code/feature summary) to see how your RTL manga folder-specific setting works? I think there are many potential applications for folder-specific settings.

**Comment by ryanwwest** [comment]
Nevermind, I see how Tweak Document Settings works. These discussions always remind me of VS Code's settings hierarchy of user, workspace, folder that increasingly override one another, but I know that's too big for the moment :)

**Comment by mergen3107** [comment]
@ryanwwest 
https://[koreader.rocks/user_guide/#manga](https://koreader.rocks/user_guide/#manga)
See `Reversing the page turn direction for authentic Japanese manga`.

Good thing about it, is that it does not write these settings to the .sdr metadata, but append during the opening of the book. So, if you move the book out, it will stay in its untouched condition.

In this suggestion I was thinking about something like this in history processing (pseudo code):
```
if tweak_document_setting contain ignore_history then
    return end
else
    add to history as usually
end
```

**Comment by ryanwwest** [comment]
Thanks, that seems simple and a good solution (though the naming of 'Tweak Document Setting' could perhaps be made clearer to indicate folder-specific settings, I didn't know about it before).

> Good thing about it, is that it does not write these settings to the .sdr metadata, but append during the opening of the book. So, if you move the book out, it will stay in its untouched condition.

Just so I understand, since these settings aren't book-specific (e.g. in .sdr metadata) you're saying that if you move a book out of the applicable folder then open it, it _will_ be saved to history / have its metadata saved / not RTL, or i.e. whatever folder-specific settings applied are no longer applied? As that's what I'd expect.

**Comment by ryanwwest** [comment]
(Here's my use case for folder-settings for both don't save to history and don't save metadata: https://github.com/gildas-lormeau/SingleFile/discussions/1353)

**Comment by mergen3107** [comment]
> Just so I understand, since these settings aren't book-specific (e.g. in .sdr metadata) you're saying that if you move a book out of the applicable folder then open it, it _will_ be saved to history / have its metadata saved / not RTL, or i.e. whatever folder-specific settings applied are no longer applied? As that's what I'd expect.

Yes, they do not write to settings, but rather append these setting on startup. You explained it correctly, it is flexible.

**Comment by poire-z** [comment]
About not adding some documents to History & statistics & not creating docsettings, there's been some discussions and user patches bits in #10308.

> A bit too niche for us to spend time thinking about a UI to allow flagging/managing/displaying such directories - and as you both agree, users may want some things or some others disabled/changed, so again hard to propose via the UI.
So, another good candidate for [user patches](https://github.com/koreader/koreader/wiki/User-patches).
